core/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -288,55 +281,75 @@ public boolean isSuggestOnly() {
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         searchType = SearchType.fromId(in.readByte());
-
         indices = new String[in.readVInt()];
         for (int i = 0; i < indices.length; i++) {
             indices[i] = in.readString();
         }
-
         routing = in.readOptionalString();
         preference = in.readOptionalString();
-
         if (in.readBoolean()) {
```

Comment:
I guess you could convert it to Writeable and give it a stream taking constructor. Then it'd make sense that it is public at least. But that shouldn't block this fix.